Group failing spec by error message

safafa opened the pull request:
## Description:

This PR adds a custom rspec formatter, allows us to detect failing specs and group them by exception. the formatter prints the results in a text file.

## How to test: 

To test this on a rails project:
Add the gem to the Gemfile then run the rspec command using these flags: 
`bundle exec rspec --require failing_spec_detector/failing_spec_formatter.rb  --format FailingSpecDetector::FailingSpecFormatter`

##Example:
<img width="1196" alt="Screenshot 2023-12-04 at 10 46 49" src="https://github.com/nebulab/failing_spec_detector/assets/43698511/c0a0ea18-7e9e-4129-a3a0-20bc8c6959e3">

safafa committed:
Update gemfile and gemspec

safafa committed:
Add failing spec formatter

DanielePalombo reviewed (commented):
Do you mind updating the README with the instruction to configure the formatted?

safafa committed:
Update Readme

MassimilianoLattanzio reviewed (approved):
Everything looks fine to me!

Later I'll try it in my application.

DanielePalombo reviewed (changes requested):
This is not published on rubygems yet, do you mind writing the right installation instructions?

```ruby
gem 'failing_spec_detector', github: 'nebulab/failing_spec_detector'
```

What do you think adding an example(txt or image) of the produced result?

safafa commented:
@DanielePalombo [here](https://github.com/nebulab/failing_spec_detector/commit/f6982c7e119863ae48fb24f572c3f1fa2795c364) I updated the Readme. I also added an example screenshot in the PR comment. Or are you suggesting adding an example in the Readme file?

DanielePalombo commented:
> @DanielePalombo [here](https://github.com/nebulab/failing_spec_detector/commit/f6982c7e119863ae48fb24f572c3f1fa2795c364) I updated the Readme. I also added an example screenshot in the PR comment. Or are you suggesting adding an example in the Readme file?

In the README file it would be better so the purpose and the result will be more clear

safafa commented:
> > @DanielePalombo [here](https://github.com/nebulab/failing_spec_detector/commit/f6982c7e119863ae48fb24f572c3f1fa2795c364) I updated the Readme. I also added an example screenshot in the PR comment. Or are you suggesting adding an example in the Readme file?
> 
> In the README file it would be better so the purpose and the result will be more clear

[Done ](https://github.com/nebulab/failing_spec_detector/commit/eec6e0d30e4ff44a25372eaabd888a14f8f14d3f), wdyt?
